File: tests/compile-fail/ForeignNullableGenericOverloadedGenericReturn.cs
using System;
using System.Collections.Generic;

// A GENERIC .NET method in an overload set. Name + generic arity + parameter count leaves both members standing,
// and stamping a fake `void` for "could not narrow it" made the crossing refusal see a member with no problematic
// return at all — while emission, which links by the exact `memberSig` the frontend resolved, picked the right one
// and handed back a `List<Nullable<int32>>` consumed as a `List<object>`.
namespace ovgen
{
    public static class Fac
    {
        public static List<int?> Make<T>(int x) => new List<int?> { x, null };

        // The sibling that shares name, generic arity and parameter count, and crosses nothing. It is driven in
        // tests/interop so the refusal is shown to discriminate rather than to fire on the set.
        public static string Make<T>(string x) => "s:" + x;
    }
}

File: tests/compile-fail/ForeignNullableGenericOverloadedGenericReturn.kt
// #86 — an OVERLOADED generic .NET method whose declared return crosses.
//
// The chokepoint asks whether a stamp was made, not whether it said anything, so writing `void` for an overload set
// this could not narrow satisfied it and silently emptied the refusal's input. The call carries the exact
// frontend-resolved declaration parameters, so bir2cir selects the same scalar memberRef and sees the crossing.
import ovgen.Fac

fun main() {
    val xs = Fac.Make<Int>(1)
    println(xs.Count)
}
